shared/feature/feedings/src/main/java/com/epmedu/animeal/feedings/presentation/model/FeedingsViewModel.kt

```diff
@@ -0,0 +1,67 @@
+package com.epmedu.animeal.feedings.presentation.model
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.epmedu.animeal.common.presentation.viewmodel.delegate.ActionDelegate
+import com.epmedu.animeal.common.presentation.viewmodel.delegate.DefaultStateDelegate
+import com.epmedu.animeal.common.presentation.viewmodel.delegate.StateDelegate
+import com.epmedu.animeal.feeding.domain.usecase.GetFeedingInProgressUseCase
+import com.epmedu.animeal.feedings.domain.usecase.GetFeedingHistoriesUseCase
+import com.epmedu.animeal.feedings.domain.usecase.GetFeedingPointsUseCase
+import com.epmedu.animeal.feedings.presentation.FeedingsScreenEvent
+import com.epmedu.animeal.feedings.presentation.viewmodel.FeedingsState
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.collections.immutable.toImmutableList
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+internal class FeedingsViewModel @Inject constructor(
+    actionDelegate: ActionDelegate,
+    private val getFeedingPointsUseCase: GetFeedingPointsUseCase,
+    private val getFeedingInProgressUseCase: GetFeedingInProgressUseCase,
+    private val getFeedingHistoriesUseCase: GetFeedingHistoriesUseCase
+) : ViewModel(),
+    StateDelegate<FeedingsState> by DefaultStateDelegate(initialState = FeedingsState()),
+    ActionDelegate by actionDelegate {
+
+    init {
+        viewModelScope.launch { collectFeedingPoints() }
+    }
+
+    fun handleEvents(event: FeedingsScreenEvent) {
+        when (event) {
+            is FeedingsScreenEvent.FeedingApprove -> {
+                // TODO: implement me
+            }
+            is FeedingsScreenEvent.FeedingReject -> {
+                // TODO: implement me
+            }
+        }
+    }
+
+    private suspend fun collectFeedingPoints() {
+        getFeedingPointsUseCase().collect { feedingPoints ->
```

Comment:
As soon as you create separate methods for getting all feeding histories and feedings in progress without the filter for feeding point id, you will be able to remove this